### DISCUSSION TITLE
How publish to only one organisation package repository

### DISCUSSION THREAD

**Discussion by pkernevez** [community/community#26078] [Actions]
<p>Hi,</p>
<p>I’m admin (and owner) of several organisations.<br>
I’m setting up the workflow for one repository of one organisation.<br>
The only way I found to publish (with maven) in one repository was to generate a personal token and used my own account.<br>
This Token have limited OAuth scopes, but I don’t find a way to restrict the token to an organisation/repository.</p>
<p>Is there a way to use a technical account and token to publish with a limited scope (OAuth &amp; repo/orga) ? I didn’t succeed to use the GITHUB_TOKEN secret by following the docs I found on this subject.</p>
<p>If not possible, is there a way to limit my personal key to a subset of my own repository ?</p>
<p>Thanks for your help,<br>
Philippe</p>
<p>PS : it’s my first experience with Action &amp; Workflow, I’m really more familiar with Gitlab concepts</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/pkernevez">@pkernevez</a>,</p>
<p>When you create a personal access token (PAT), there is not method to limit the scope of this PAT to a specific organization or repository. What you can choose is to limit the scope to private repositories or public repositories or all repositories on your organisations and user account. More details, see “<a href="https://docs.github.com/en/developers/apps/scopes-for-oauth-apps#available-scopes" rel="nofollow noopener"><strong>Available scopes</strong></a>”.</p>
<p><a href="https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#about-the-github_token-secret" rel="nofollow noopener">The permissions of the <strong>GITHUB_TOKEN</strong> are limited to the repository that contains your workflow.</a><br>
That means the <strong>GITHUB_TOKEN</strong> is only available to the repository where the workflow is running. You can’t use it to access other repositories in the workflow.</p>

**Comment by pkernevez** [comment]
<p>Thanks for your response.</p>
<p>I finally found the right way to use the github_token, the documentation purpose many ways but not of them are right (or complete in fact).</p>
<p>I had to:</p>
<ol>
<li>Defined a settings.xml file:</li>
</ol>
<pre><code class="lang-auto">    &lt;servers&gt;
        &lt;server&gt;
            &lt;id&gt;github_release&lt;/id&gt;
            &lt;configuration&gt;
                &lt;httpHeaders&gt;
                    &lt;property&gt;
                        &lt;name&gt;Authorization&lt;/name&gt;
                        &lt;value&gt;Bearer ${env.GITHUB_TOKEN}&lt;/value&gt;
                    &lt;/property&gt;
                &lt;/httpHeaders&gt;
            &lt;/configuration&gt;
        &lt;/server&gt;
    &lt;/servers&gt;
&lt;/settings&gt;
</code></pre>
<ol start="2">
<li>Use the right distribution repositories (and replace $org and $repo with your real organisation and repository:</li>
</ol>
<pre><code class="lang-auto">    &lt;distributionManagement&gt;
        &lt;repository&gt;
            &lt;id&gt;github_release&lt;/id&gt;
            &lt;url&gt;https://maven.pkg.github.com/$org/$repo&lt;/url&gt;
        &lt;/repository&gt;
        &lt;snapshotRepository&gt;
            &lt;id&gt;github_release&lt;/id&gt;
            &lt;url&gt;https://maven.pkg.github.com/$org/$repo&lt;/url&gt;
        &lt;/snapshotRepository&gt;
    &lt;/distributionManagement&gt;
</code></pre>
<ol start="3">
<li>Use the right parameter to maven command action:</li>
</ol>
<pre><code class="lang-auto">    - name: Publish to GitHub Packages Apache Maven
      run: mvn -s ${GITHUB_WORKSPACE}/maven/settings.xml -B deploy
      working-directory: java
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        GITHUB_WORKSPACE: ${{ github.workspace }}
</code></pre>
